k2/csrc/array.h

```diff
@@ -0,0 +1,98 @@
+// k2/csrc/array.h
+
+// Copyright (c)  2020  Xiaomi Corporation (author: Daniel Povey)
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#ifndef K2_CSRC_ARRAY_H_
+#define K2_CSRC_ARRAY_H_
+
+#include <functional>
+#include <limits>
+#include <memory>
+#include <vector>
+
+namespace k2 {
+
+
+/*
+   We will use e.g. StridedPtr<int32_t, T> when the stride is not 1, and otherwise
+   just T* (which presumably be faster).
+*/
+template <typename I, typename T>
+class StridedPtr {
+  T *data;
+  I stride;
```

Comment:
Does the `stride` designate the number of bytes or number of elements?

Reply:
I was thinking number of elements.  We could change it later, of course, if bytes was more convenient.